CMakeLists.txt

```diff
@@ -69,27 +71,38 @@ if(nanopb_BUILD_GENERATOR)
         install(
             FILES ${PROJECT_BINARY_DIR}/${generator_proto_py_file}
                   ${generator_proto_file}
-            DESTINATION ${PYTHON_INSTDIR}/proto/
+            DESTINATION ${PYTHON_INSTDIR}/nanopb/generator/proto/
         )
     endforeach()
 
     install( FILES generator/proto/_utils.py
                    generator/proto/__init__.py
-             DESTINATION ${PYTHON_INSTDIR}/proto/ )
+             DESTINATION ${PYTHON_INSTDIR}/nanopb/generator/proto/ )
+
+    install( FILES generator/nanopb_generator.py
+                   generator/__init__.py
+             DESTINATION ${PYTHON_INSTDIR}/nanopb/generator/ )
+
+    install( FILES generator/__init__.py
+             DESTINATION ${PYTHON_INSTDIR}/nanopb/ )
+
 endif()
 
+# Install small script wrappers to invoke the generator from the installed module
 if(WIN32)
     install(
         PROGRAMS
-            generator/nanopb_generator.py
+            extra/script_wrappers/nanopb_generator.py
             generator/protoc-gen-nanopb.bat
+            generator/nanopb_generator.bat
         DESTINATION ${CMAKE_INSTALL_BINDIR}
     )
 else()
     install(
         PROGRAMS
-            generator/nanopb_generator.py
+            extra/script_wrappers/nanopb_generator.py
             generator/protoc-gen-nanopb
+            generator/nanopb_generator
         DESTINATION ${CMAKE_INSTALL_BINDIR}
```

Comment:
Maybe you should move the common files into its own install command.

Reply:
You mean `nanopb_generator.py`? Possibly, though if it is just the single file it may not make the result any simpler.

Reply:
It would make clear that this is a common file to resolve the local imports to the installed package.

Reply:
True. I reordered the commands now.